src/Plugin/AccessControlHierarchy/Taxonomy.php

```diff
@@ -353,9 +356,39 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
       '#options' => $taxonomy_fields,
       '#default_value' => array_combine($default_value, $default_value),
     ];
+    $form['validate'] = ['#type' => 'value', '#value' => $validate];
     return $form;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $settings = $form_state->getValues();
+    $validate = $settings['validate'];
+    $settings['vocabularies'] = array_values(array_filter($settings['vocabularies']));
+    $settings['fields'] = array_values(array_map(function ($item) {
+      list($entity_type, $bundle, $field_name) = explode(':', $item);
+      return [
+        'entity_type' => $entity_type,
+        'bundle' => $bundle,
+        'field' => $field_name,
+      ];
+    }, array_filter($settings['fields'])));
+    foreach ($settings['validate'] as $index => $value) {
+      $error = TRUE;
+      foreach ($settings['vocabularies'] as $vocabulary) {
+        if (in_array($vocabulary, $value)) {
+          $error = FALSE;
+        }
+      }
+      if ($error) {
+        $field = $form['fields']['#options'][$index];
+        $form_state->setErrorByName('scheme_settings][fields][' . $index, $this->t('The field @field on content type @type is not in the selected vocabularies.', ['@field' => $field['field'], '@type' => $field['bundle']]));
```

Comment:
Should we add a test for this?

Reply:
We should. It's fairly easy to do, on tap for tomorrow.